docs/config-definition.md

```diff
@@ -50,7 +45,7 @@ The configurable parameters are as follows:
 
 | Key | Type | Default | Meaning |
 | --- | --- | --- | --- |
-| `persistentVolumeProvider` | CloudProviderConfig<br><br>(Supported key values are `aws`, `gcp`, and `azure`, but only one can be present. See the corresponding [AWS][0], [GCP][1], and [Azure][2]-specific configs.) | Required Field | The specification for whichever cloud provider the cluster is using for persistent volumes (to be snapshotted).<br><br> *NOTE*: For Azure, your Kubernetes cluster needs to be version 1.7.2+ in order to support PV snapshotting of its managed disks. |
+| `persistentVolumeProvider` | CloudProviderConfig<br><br>(Supported key values are `aws`, `gcp`, and `azure`, but only one can be present. See the corresponding [AWS][0], [GCP][1], and [Azure][2]-specific configs.) | None (Optional) | The specification for whichever cloud provider the cluster is using for persistent volumes (to be snapshotted), if any.<br><br>If not specified, Backups and Restores requesting PV snapshots & restores, respectively, are considered invalid. <br><br> *NOTE*: For Azure, your Kubernetes cluster needs to be version 1.7.2+ in order to support PV snapshotting of its managed disks. |
```

Comment:
@skriss is the primary motivation for making PVProvider optional to support running non-cloud (i.e. minio)?

I do think @abiogenesis-now brings up a good point. As far as UX goes, I think having a backup/restore fail validation because you forgot to set `--snapshot-volumes=false` might get annoying after a while, if the server isn't configured appropriately. I guess we might need to decide how likely running without a PV provider is. Also need to decide if we need to distinguish between something like "minio and legitimately no PV provider" and "aws/gcp/azure and I forgot to configure the PV provider, so it should be a validation error". Let's continue to discuss...